cli/status/status.go

```diff
@@ -1,57 +1,185 @@
 package status
 
 import (
+	_ "embed"
+	"fmt"
 	"os"
+	"strings"
 
+	"github.com/fatih/color"
 	"github.com/jedib0t/go-pretty/v6/table"
 	"github.com/jedib0t/go-pretty/v6/text"
+	"github.com/mitchellh/mapstructure"
 	"github.com/tarantool/tt/cli/connector"
 	"github.com/tarantool/tt/cli/process_utils"
 	"github.com/tarantool/tt/cli/running"
 )
 
+//go:embed lua/instance_state.lua
+var replicationInfoLuaScript string
+
 // StatusOpts contains options for tt status.
 type StatusOpts struct {
 	// Option for pretty-formatted table output.
-	Pretty bool
+	Pretty  bool
+	Details bool
+}
+
+type Alert struct {
+	Type    string `mapstructure:"type"`
+	Message string `mapstructure:"message"`
+}
+
+type ConfigInfo struct {
+	Status string  `mapstructure:"status"`
+	Alerts []Alert `mapstructure:"alerts"`
+}
+
+type Upstream struct {
+	Status  string `mapstructure:"status"`
+	Message string `mapstructure:"message"`
+}
+
+type ReplicationInfo struct {
+	Name     string    `mapstructure:"name"`
+	Upstream *Upstream `mapstructure:"upstream"`
+}
+
+type InstanceState struct {
+	ReplicationInfo []ReplicationInfo `mapstructure:"replication_info"`
+	ConfigInfo      ConfigInfo        `mapstructure:"config_info"`
+	ReadOnly        string            `mapstructure:"read_only"`
+	BoxStatus       string            `mapstructure:"box_status"`
 }
 
 // Status writes the status as a table.
 func Status(runningCtx running.RunningCtx, opts StatusOpts) error {
 	ts := table.NewWriter()
 	ts.SetOutputMirror(os.Stdout)
-	ts.AppendHeader(table.Row{"INSTANCE", "STATUS", "PID", "MODE"})
+	ts.AppendHeader(
+		table.Row{"INSTANCE", "STATUS", "PID", "MODE", "CONFIG", "BOX", "UPSTREAM"})
+
+	instances := map[string]map[string]interface{}{}
+	instancesAlerts := map[string][]string{}
+
+	PrintYellow := color.New(color.FgYellow).SprintFunc()
+	PrintRed := color.New(color.FgRed).SprintFunc()
 
 	for _, run := range runningCtx.Instances {
-		row := []interface{}{}
 		fullInstanceName := running.GetAppInstanceName(run)
 		procStatus := running.Status(&run)
 
-		row = append(row, fullInstanceName)
-		row = append(row, procStatus.ColorSprint(procStatus.Status))
+		if _, exists := instances[fullInstanceName]; !exists {
+			instances[fullInstanceName] = map[string]interface{}{
+				"STATUS": "",
+				"PID":    nil,
+				"MODE":   "",
+				"CFG":    "--",
+				"BOX":    "--",
+				"UPS":    "--",
+			}
```

Comment:
It feels a bit inconsistent that we have `CONFIG` column in the output and `CFG` field here, the same for `UPSTREAM` and `UPS`.

Also, shouldn't we use a structure here instead of a map?

Reply:
Renamed.
I don't prefer to use structure here because we can't change the structure in the map directly. We have to do something like this:
```go
s := map[name]
s.field_1 = ...
s.field_2 = ...
map[name] = s
```

Reply:
OK, that's just thoughts around. I'll leave all the details about the implementation to @oleg-jukovec and @psergee. Closing this discussion.